File: telegram-bot-core/src/main/java/ru/tggc/telegrambotcore/registry/resolver/UsernameResolver.java
package ru.tggc.telegrambotcore.registry.resolver;

import org.springframework.stereotype.Component;
import ru.tggc.telegrambotcore.annotation.params.Username;

import java.lang.reflect.Parameter;

@Component
public class UsernameResolver implements ParameterResolver<String> {

    @Override
    public boolean supports(Parameter p) {
        return p.isAnnotationPresent(Username.class);
    }

    @Override
    public String resolve(Parameter parameter, HandlerCtx ctx) {
        return ctx.from().username();
    }
}

File: telegram-bot-core/src/main/kotlin/ru/tggc/telegrambotcore/registry/resolver/UsernameResolver.kt
package ru.tggc.telegrambotcore.registry.resolver

import org.springframework.stereotype.Component
import ru.tggc.telegrambotcore.annotation.params.Username
import java.lang.reflect.Parameter

@Component
class UsernameResolver : ParameterResolver<String?> {
    override fun supports(p: Parameter?): Boolean? = p?.isAnnotationPresent(Username::class.java)

    override fun resolve(parameter: Parameter?, ctx: HandlerCtx): String? = ctx.from!!.username()
}
